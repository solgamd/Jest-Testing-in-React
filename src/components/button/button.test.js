import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../button/button';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button></Button>, div);
});

it('renders button correctly', () => {
    const { getByTestId } = render(<Button label="Click me"></Button>);
    expect(getByTestId('button')).toHaveTextContent('Click me');
});

it('renders button correctly', () => {
    const { getByTestId } = render(<Button label="Save"></Button>);
    expect(getByTestId('button')).toHaveTextContent('Save');
});

it('matches snapshot 1', () => {
    const tree = renderer.create(<Button label="Save"></Button>).toJSON();
    expect(tree).toMatchSnapshot(); //looks at file tree for 'button.snapshot" (or creates it) and dumps entire JSON into it
});

it('matches snapshot 2', () => {
    const tree = renderer.create(<Button label="Click me"></Button>).toJSON();
    expect(tree).toMatchSnapshot(); //looks at file tree for 'button.snapshot" (or creates it) and dumps entire JSON into it
});