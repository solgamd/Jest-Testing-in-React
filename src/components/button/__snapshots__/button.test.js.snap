// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="btn-style btn"
  data-testid="button"
>
  Save
</div>
`;

exports[`matches snapshot 1 1`] = `
<div
  className="btn-style btn"
  data-testid="button"
>
  Save
</div>
`;
